flask_appbuilder/tests/_test_auth_ldap.py

```diff
@@ -320,6 +340,30 @@ def test__inactive_user(self):
         # validate - expected LDAP methods were called
         self.assertEquals(self.ldapobj.methods_called(with_args=True), [])
 
+    def test__multi_group_user_mapping_to_same_role(self):
+        """
+        LDAP: test login flow for - user in multiple groups mapping to same role
+        """
+        self.app.config["AUTH_ROLES_MAPPING"] = {
+            "cn=staff,ou=groups,o=test": ["Admin"],
+            "cn=admin,ou=groups,o=test": ["Admin", "User"],
+        }
+        self.app.config["AUTH_LDAP_SEARCH"] = "ou=users,o=test"
+        self.app.config["AUTH_LDAP_USERNAME_FORMAT"] = "uid=%s,ou=users,o=test"
+        self.app.config["AUTH_USER_REGISTRATION"] = True
+        self.app.config["AUTH_USER_REGISTRATION_ROLE"] = "Public"
+        self.appbuilder = AppBuilder(self.app, self.db.session)
+        sm = self.appbuilder.sm
+
+        # validate - no users are registered
+        self.assertEqual(sm.get_all_users(), [])
+
+        # attempt login
+        user = sm.auth_user_ldap("natalie", "natalie_password")
+
+        # validate - user was allowed to log in
+        self.assertIsInstance(user, sm.user_model)
```

Comment:
After this can you add the following checks after this:
1. "user was registered" (e.g. [link](https://github.com/dpgaspar/Flask-AppBuilder/blob/v3.3.0/flask_appbuilder/tests/_test_auth_ldap.py#L746-L747))
2. "user was given the correct roles" (e.g. [link](https://github.com/dpgaspar/Flask-AppBuilder/blob/v3.3.0/flask_appbuilder/tests/_test_auth_ldap.py#L749-L753))
3. "user was given the correct attributes (read from LDAP)" (e.g. [link](https://github.com/dpgaspar/Flask-AppBuilder/blob/v3.3.0/flask_appbuilder/tests/_test_auth_ldap.py#L755-L758))
4. "expected LDAP methods were called" (e.g. [link](https://github.com/dpgaspar/Flask-AppBuilder/blob/v3.3.0/flask_appbuilder/tests/_test_auth_ldap.py#L760-L769))